main.go

```diff
@@ -68,36 +93,49 @@ func run() error {
 	return app.Run(os.Args)
 }
 
+func parseArgs(ctx *cli.Context, f string) (*os.File, *string, error) {
+	aa := ctx.Args()
+	if aa.Len() != 1 {
+		return nil, nil, cli.ShowCommandHelp(ctx, f)
+	}
+
+	specPath := aa.First()
+	if _, err := os.Stat(specPath); err != nil {
+		return nil, nil, err
+	}
+
+	r, err := os.Open(specPath)
+	return r, &specPath, err
+}
+
 func generateDocumentation(f generator.Format) func(ctx *cli.Context) error {
 	return func(ctx *cli.Context) error {
-		aa := ctx.Args()
-		if aa.Len() != 1 {
-			return cli.ShowCommandHelp(ctx, string(f))
+		r, specPath, err := parseArgs(ctx, string(f))
+		if err != nil {
+			return err
 		}
-
-		specPath := aa.First()
-		if _, err := os.Stat(specPath); err != nil {
+		g, err := generator.New(f)
+		if err != nil {
 			return err
 		}
+		return g.Generate(ctx.Context, *specPath, r, ctx.App.Writer)
+	}
+}
 
-		r, err := os.Open(specPath)
+func convert(f converter.Format) func(ctx *cli.Context) error {
+	return func(ctx *cli.Context) error {
+		r, specPath, err := parseArgs(ctx, string(f))
 		if err != nil {
 			return err
 		}
-
-		g, err := generator.New(f)
+		c, err := converter.New(f)
 		if err != nil {
 			return err
 		}
-
-		return g.Generate(ctx.Context, specPath, r, ctx.App.Writer)
+		return c.Convert(ctx.Context, *specPath, r, ctx.App.Writer)
```

Comment:
I understand the code is made by analogy with the old code. But it seems that it will be easier if we receive bytes here and immediately write them to ctx.App.Writer. If we don’t pass ctx.App.Writer somewhere further.

Reply:
The code turned out to be confusing. I think it can be simplified here.

Reply:
let's set the application version to 2.0.0

Reply:
fixed, so version is 1.2.0 now